### PR TITLE
added variable for id token user property

### LINKED ISSUE #889: User can modify the id token property to link to Conjur username
Another variable called "IDTokenUserProperty" will be set with the name of the property in the ID Token by which we will map the user name in Conjur.

If the variable doesn’t exist, or it has no value, the app should fail as we do with other oidc vars

### PR DESCRIPTION
**Pull request by orenbm**
#### What does this PR do?
the user property for ID token is configurable via a variable

#### Any background context you want to provide?
we want to let the user configure the property from the id token with which she'd like to map to a conjur username

#### What ticket does this PR close?
closes #889

### CODE REVIEW

File: authn-oidc-files/policy.yml [hunk 1]
```diff
@@ -14,6 +14,9 @@
   - !variable
     id: provider-uri
 
+  - !variable
```
Comment (InbalZilberman) [inline authn-oidc-files/policy.yml]: I believe this var is defined twice in the same policy

File: app/domain/authentication/authn_oidc/authenticate.rb [hunk 2]
```diff
@@ -53,8 +58,7 @@ def new_token(input)
         )
       end
 
-      def conjur_username(request_body)
-        id_token_username_field = "preferred_username"
+      def conjur_username(request_body, id_token_username_field)
         conjur_username = request_body.id_token[id_token_username_field]
         raise IdTokenFieldNotFound, id_token_username_field unless conjur_username&.present?
```
Comment (InbalZilberman) [inline app/domain/authentication/authn_oidc/authenticate.rb]: As I am not really familiar with the code..  I maybe asking a dummy question
the logic of this POC is only for conjur users that are located in root (and root only) where in the code do you make this assumption?

File: authn-oidc-files/policy.yml (hunk 1 above)
Comment (orenbm) [reply authn-oidc-files/policy.yml]: Yes, we have 2 policies in this file - one for okta and one for keycloak

File: app/domain/authentication/authn_oidc/authenticate.rb (hunk 2 above)
Comment (orenbm) [reply app/domain/authentication/authn_oidc/authenticate.rb]: we don't. why should we limit this in the code?

Comment (orenbm) [reply app/domain/authentication/authn_oidc/authenticate.rb]: it is a limitation we have by design

File: cucumber/authenticators/features/step_definitions/authn_common_steps.rb [hunk 3]
```diff
@@ -1,3 +1,5 @@
+require_relative '../../../../cucumber/api/features/step_definitions/authz_steps'
```
Comment (jtuttle) [inline cucumber/authenticators/features/step_definitions/authn_common_steps.rb]: Are you only using a single step definition from this file? If so, I would just duplicate it here and then add a TODO comment and a GitHub issue that say something about finding a good way to share step definitions between different cucumber categories.

Comment (orenbm) [reply cucumber/authenticators/features/step_definitions/authn_common_steps.rb]: done

**Pull request merged**